internal/locate/cse/fallback.go

```diff
@@ -0,0 +1,126 @@
+package cse
+
+import (
+	"context"
+	"time"
+
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/log"
+	pd "github.com/tikv/pd/client"
+	"go.uber.org/zap"
+)
+
+var (
+	_ pd.Client = &Fallback{}
+)
+
+type Fallback struct {
+	pd.Client
+	cse     pd.Client
+	breaker *asyncBreaker
+}
+
+func probePD(name string, client pd.Client, timeout time.Duration) error {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	_, err := client.GetRegionByID(ctx, 1)
```

Comment:
I try to probe PD with GetRegionByID because the health status of PD is hard to define since there will be more services in the PD microservice. We just probe the PD with the API we care about, a.k.a the region meta service.